Source/WebCore/accessibility/AccessibilityObject.cpp

```diff
@@ -1027,8 +1054,19 @@ Vector<String> AccessibilityObject::performTextOperation(const AccessibilityText
     size_t replacementStringsCount = operation.replacementStrings.size();
     bool useFirstReplacementStringForAllReplacements = (replacementStringsCount == 1);
 
-    for (size_t i = 0; i < operation.textRanges.size(); ++i) {
-        auto textRange = operation.textRanges[i];
+    for (const auto& range : operation.textRanges) {
+        auto textOperationRange = textOperationRangeFromRange(range);
+        if (!textOperationRange) {
+            ASSERT_NOT_REACHED();
+            return result;
+        }
+
+        textOperationRanges.append(*textOperationRange);
+    }
```

Comment:
Performing a replacement can invalidate other stored SimpleRanges, which prevents us from performing a selection, causing this bug.

That is why I am constructing the character ranges while the start/end containers of the stored SimpleRanges are still valid (i.e. **before** performing any replacement).

The call to `rootEditableElement` would fail for all orphaned range start/end containers if I attempted to construct the character range in the same loop as where I perform the replacement.

I tried to address this in the commit message:

>To address this issue, before performing any replacements, we construct character ranges with
>respect to their root editable element for all of the provided ranges. The root editable element
>seems to be preserved in the tree after performing a replacement using the `ReplaceSelectionCommand`.
>Then, we can reconstruct the `SimpleRange` for each replacement, ensuring we always use a valid
>SimpleRange.

Reply:
Ahh, right, thanks! Would you please add a comment explaining the reason for the separate loops so the reason is obvious directly in the code?

Reply:
Added a comment in code :)